### ISSUE TITLE
Support for multiple host

### ISSUE THREAD

**Issue by wzshiming** [distribution/distribution#3553]
Can it support using different `manifests` according to the `Host` in the `Header`, but using the same `blobs`

**Comment by brackendawson** [comment]
That's not currently achievable using distribution. The registry is largely unaware of its own domain name. Nether manifests nor blobs are stored under any host information, so pointing two domains at one distribution registry causes the images to be aliases for one another. People already make use of this, for example `registry.ng.bluemix.net/bracken/image:latest` is the same as `us.icr.io/bracken/image:latest`, any changes would need to not break that feature.

The easiest way to achieve this would probably be a new storage driver which is a shallow wrapper for the existing drivers, but which allows the manifest requests to be directed to multiple back ends based on host, while the blob requests are all sent to one back end. But you would need to get the host information into the storage calls, that might be a very big change.

Another approach that would work with an unmodified distribution release would be to re-write the path in your ingress controller to add the host as the first part of the image name, so `some.example.com/v2/image/manifests/latest` would become `example.com/v2/some.example.com/image/manifests/latest`. Your auth mechanism would need to be aware of this though.

**Comment by wzshiming** [comment]
Thank you for response

> Another approach that would work with an unmodified distribution release would be to re-write the path in your ingress controller to add the host as the first part of the image name, so some.example.com/v2/image/manifests/latest would become example.com/v2/some.example.com/image/manifests/latest.
I have tried this scheme, there is layer Nginx to do URI conversion and Forward, there seems to be a problem with uploading image.

I also tried mounting blobs in the container to the same host folder, An error message is reported when uploading files, and cross-drive movement is not supported.
https://github.com/distribution/distribution/blob/1563384b69df9376389fe45ce949173a6383770a/registry/storage/driver/filesystem/driver.go#L267


I feel that the most likely way to achieve this is to modify storage driver. 
My idea is to add a configuration field `hosts` to all drivers, For requests that the Host domain name is included in `hosts`, add host when splicing the manifests path.
